bottlecap/src/secrets/decrypt.rs

```diff
@@ -0,0 +1,93 @@
+use std::io::Error;
+use crate::config::Config;
+use aws_config;
+use aws_sdk_secretsmanager::Client;
+use aws_sdk_secretsmanager;
+use tracing::debug;
+use std::thread;
+
+
+type ResolveFn = fn(String) -> Result<String, Error>;
+
+pub fn resolve_secrets(config: Config, decoder: ResolveFn) -> Result<Config, String> {
+    if !config.api_key.is_empty() {
+        debug!("DD_API_KEY found, not trying to resolve secrets");
+        Ok(config)
+    } else {
+        if !config.api_key_secret_arn.is_empty() {
+            debug!("DD_API_KEY_SECRET_ARN found, trying to resolve ARN secret");
+            let string = config.api_key_secret_arn.clone();
+            let wait_fot_ret = thread::spawn(move || { decoder(string) });
```

Comment:
```suggestion
            let wait_for_ret = thread::spawn(move || { decoder(string) });
```
what does `ret` mean?

Reply:
wait for return, but this code is definitely useless. There was a lot of back and forth and I will refactor before next submission